pkg/util/parameters.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package util
+
+import (
+	"fmt"
+
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	"github.com/samber/lo"
+)
+
+var (
+	ErrUnknownParameter      = fmt.Errorf("unknown parameter")
+	ErrUnmodifiableParameter = fmt.Errorf("parameter is not modifiable")
+)
+
+// Parameters represents the elements of a DB Parameter Group
+// or a DB Cluster Parameter Group
+type Parameters map[string]*string
```

Comment:
Type aliasing for basic Go type is very rarely recommended in idiomatic Go - i'd suggest just use `map[string]*string` in the function parameters

Reply:
https://github.com/aws-controllers-k8s/runtime/blob/fb553741e0bfbbd1fee8f52b2e36b8c58dd7bf8e/pkg/tags/tags.go#L21

My intent was to create an analog to the [GetTagsDifference](https://github.com/aws-controllers-k8s/runtime/blob/7152103f915b2722bef679ebe6df231e7f489222/pkg/compare/tags.go#L26) function

as [suggested here](https://github.com/aws-controllers-k8s/rds-controller/pull/133#discussion_r1130237940)